src/cool_seq_tool/mappers/exon_genomic_coords.py

```diff
@@ -521,60 +660,60 @@ async def _get_alt_ac_start_and_end(
         transcript: str | None = None,
         gene: str | None = None,
         get_nearest_transcript_junction: bool = False,
         is_start: bool = True,
-    ) -> TranscriptExonDataResponse:
-        """Convert individual genomic data to transcript data
+    ) -> _GenomicTxSeg:
+        """Given genomic data, generate a boundary for a transcript segment.
 
         :param genomic_pos: Genomic position where the transcript segment starts or ends
             (inter-residue based)
         :param chromosome: Chromosome. Must give chromosome without a prefix
-            (i.e. ``1`` or ``X``). If not provided, must provide ``alt_ac``.
-            If ``alt_ac`` is also provided, ``alt_ac`` will be used.
-        :param alt_ac: Genomic accession (i.e. ``NC_000001.11``). If not provided,
-            must provide ``chromosome. If ``chromosome`` is also provided, ``alt_ac``
+            (i.e. ``1`` or ``X``). If not provided, must provide ``genomic_ac``. If
+            position maps to both GRCh37 and GRCh38, GRCh38 assembly will be used.
+            If ``genomic_ac`` is also provided, ``genomic_ac`` will be used.
+        :param genomic_ac: Genomic accession (i.e. ``NC_000001.11``). If not provided,
+            must provide ``chromosome. If ``chromosome`` is also provided, ``genomic_ac``
             will be used.
         :param transcript: The transcript to use. If this is not given, we will try the
             following transcripts: MANE Select, MANE Clinical Plus, Longest Remaining
             Compatible Transcript
         :param gene: HGNC gene symbol
         :param get_nearest_transcript_junction: If ``True``, this will return the
-            adjacent exon if the position specified by``genomic_start`` or
-            ``genomic_end`` does not occur on an exon. For the positive strand, adjacent
+            adjacent exon if the position specified by``seg_start_genomic`` or
+            ``seg_end_genomic`` does not occur on an exon. For the positive strand, adjacent
             is defined as the exon preceding the breakpoint for the 5' end and the exon
             following the breakpoint for the 3' end. For the negative strand, adjacent
             is defined as the exon following the breakpoint for the 5' end and the exon
             preceding the breakpoint for the 3' end.
         :param is_start: ``True`` if ``genomic_pos`` is where the transcript segment starts.
             ``False`` if ``genomic_pos`` is where the transcript segment ends.
-        :return: Transcript data (inter-residue coordinates)
+        :return: Data for a transcript segment boundary (inter-residue coordinates)
         """
-        resp = TranscriptExonDataResponse(
-            transcript_exon_data=None, warnings=[], service_meta=service_meta()
-        )
-        params = {key: None for key in TranscriptExonData.model_fields}
+        params = {key: None for key in _GenomicTxSeg.model_fields}
 
         if get_nearest_transcript_junction:
```

Comment:
very big method. Can we refactor it into separate methods for the two condition branches conditions (`if get_nearest_transcript_junction`,  and the rest of this method)?

Reply:
I started doing this, but we need to get something out ASAP. Will save for #224 epic

Reply:
fine with me. I think you've commented on/resolved everything I saw, feel free to squash and merge when you're ready.